pkg/validations/base.go

```diff
@@ -31,6 +33,32 @@ func RunValidations(request reconcile.Request, obj client.Object, kind string, i
 		return
 	}
 
+	// If controller has no replicas clear existing metrics and
+	// do not run any validations
+
+	objValue := reflect.Indirect(reflect.ValueOf(obj))
+	spec := objValue.FieldByName("Spec")
+	if spec.IsValid() {
+		replicas := spec.FieldByName("Replicas")
+		if replicas.IsValid() {
+			numReplicas, ok := replicas.Interface().(*int32)
+			if ok {
+				if numReplicas != nil {
```

Comment:
This technically should never happen.  I'm not sure what condition could result in a nil replicas field.  I would just treat this as if replicas are 0.  You can compress this to:

if numReplicas ** *numReplicas <= 0 {
...
} else {
...
}

Reply:
@rrati Using the change I receive an error saying `cannot indirect *numReplicas`. Since you mentioned there isn't a case where the result could be nil at that point should we just leave it as `if *numReplicas <= 0` at that point and omit the left side of the statement all together?

Reply:
> > `if numReplicas ** *numReplicas <= 0`
> 
> Is this supposed to say: `if numReplicas == nil || *numReplicas <= 0` ?

This was supposed to be `numReplicas && *numReplicas <= 0`

Reply:
> @rrati Using the change I receive an error saying `cannot indirect *numReplicas`. Since you mentioned there isn't a case where the result could be nil at that point should we just leave it as `if *numReplicas <= 0` at that point and omit the left side of the statement all together?

I'm not aware of an instance where this could be nil.  I tend to favor good defensive programming through and checking that a ptr is not nil should be done before acting upon it.

Reply:
@rrati Does the current change work? numReplicas is an int32 so needs to be compared to something for it to work. I condensed the if statement. Might not need the `ok` checked with all of that if numReplicas validity is checked.

Reply:
Considering this suggestion from Rob: 
> I'm not sure what condition could result in a nil replicas field. I would just treat this as if replicas are 0.

I think this implies we want to try to clear labels if either of the following cases are true:
* "replicas" field doesn't exist (we fail to get a value for numReplicas)
* "replicas" field exists, and is 0 or less

Therefore I propose this block of code should look like this:
```
objValue := reflect.Indirect(reflect.ValueOf(obj))
spec := objValue.FieldByName("Spec")
if spec.IsValid() {
	replicas := spec.FieldByName("Replicas")
	numReplicas, ok := replicas.Interface().(*int32)

        // clear labels if we fail get a value for numReplicas, or if value is 0 or less
	if !ok || numReplicas == nil || *numReplicas <= 0 { 
		engine.DeleteMetrics(promLabels)
		return
	}
}
```
or maybe this (whichever you find more readable, they should both do the same thing):
```
objValue := reflect.Indirect(reflect.ValueOf(obj))
spec := objValue.FieldByName("Spec")
if spec.IsValid() {
	replicas := spec.FieldByName("Replicas")
	numReplicas, ok := replicas.Interface().(*int32)

        // treat missing numReplicas value as 0
        if !ok || numReplicas == nil {
            val := 0
            *numReplicas = &val
        }

	if *numReplicas <= 0 { 
	   engine.DeleteMetrics(promLabels)
	   return
	}
}
```